### ISSUE TITLE
[sng] Not playable

### ISSUE THREAD

**Issue by rayman3003** [aargirakis/BZRPlayer#227]
Found an audio format that can't be played in BZR2.

BZR's Version: 2.0.25 Alpha

Note: I'm almost sure that they r music files, bcuz the names say it all: MUSIC1 - Tetris. & sng=song?!.

Some samples:

[sng.zip](https://github.com/aargirakis/BZRPlayer/files/9943625/sng.zip)

**Comment by ciros88** [comment]
i think this is an an Adlib Tracker module, dunno if https://github.com/Wohlstand/libADLMIDI could play it

**Comment by ciros88** [comment]
or they can be https://www.msx.org/wiki/SCC_Musixx

**Comment by ciros88** [comment]
evil.sng apart (that is an ETracker module), the other files should be Adlib Tracker but their *.ins insturment files are missing. could u provide them @rayman3003? they should have the same *.sng filename

**Comment by rayman3003** [comment]
> evil.sng apart (that is an ETracker module), the other files should be Adlib Tracker but their *.ins insturment files are missing. could u provide them [@rayman3003](https://github.com/rayman3003)? they should have the same *.sng filename

I searched for all files with the same name; But there is no *.ins can be found in all the archive (it's a 85GB archive containing 424,272 files). Also in the folder of sample files, it is only *.sng files.

Here is the screenshot:

![Image](https://github.com/user-attachments/assets/cbd3f7d6-3684-4078-b2c5-6c6d6cd52460)

**Comment by rayman3003** [comment]
> i don't know then.. do u have any software able to play them (evil.sng apart)?

Nope. I try so many apps. The only player that can play only the evil.sng, is zxtune.

> where we can find that big exotic music archive?

Here:

[ftp://modland.com/pub/modules/](url)

> in that archive, there are *.ins files somewhere? with different file names too

Yeap. I found a total of 129 ins files there. Here I upload them:

[10.zip](https://github.com/user-attachments/files/19315541/10.zip)

**Comment by ciros88** [comment]
okay so those tracks comes from modland.. from which folder exactly? normally the folder name represent the format name

**Comment by ciros88** [comment]
okey i think i found the format: https://aminet.net/package/mus/play/scplay

**Comment by rayman3003** [comment]
> from which folder exactly?

modules\Sam Coupe SNG\Frantisek Fuka

modules\Sam Coupe SNG\Ziutek

modules\Sam Coupe SNG\- unknown

**Comment by rayman3003** [comment]
> okey i think i found the format: https://aminet.net/package/mus/play/scplay

Yeap. I think thats it.

**Comment by ciros88** [comment]
so, removing the uade plugin EVIL.SNG correctly plays using zxtune.
the other sng files still dunno what they are, modland may be wrong on that, but i suppose we can exclude adlib.

**Comment by ciros88** [comment]
just to give a context, Sam Coupé was a sort of zx spectrum clone.
according to these sources:

https://aminet.net/package/mus/play/scplay
https://github.com/Deltafire/SCPlayer

Sam Coupé song files (*.sng and *.cop) embed the replay routine, or E-Tracker replayer can be used.

according to zxtune sources:
another audio format exists for Sam Coupé, it is called Aley's Module (ALM) and is also compatible with any pc xt

**Comment by ciros88** [comment]
keep in mind a coupel of things btw:
bzr2 uses a zxtune version from 2018 (r4110)
the bzr2 support for zxtune is currently very buggy
we plan to fix it (and lighten its fingerprint) or skip it and searching for a better solution, still don't know

i might have time and will to try to fix it, and thus try also to update to a more recent zxtune version, who knows

**Comment by rayman3003** [comment]
> * bzr2 uses a zxtune version from 2018 (r4110)
> * the bzr2 support for zxtune is currently very buggy
> * we might fix it (and lighten its fingerprint since it is quite big) or skip it and searching for a better solution, still don't know
> 
> i might have time and will to try to fix it, and thus try also to update to a more recent zxtune version, who knows

Yeap. Take your time. & Thank u.

**Comment by ciros88** [comment]
thanks to #91 now EVIL.SNG works but webuade has priority over it

**Comment by rayman3003** [comment]
zxtune's developer said that he is not going to support .sng of Sam Coupe E-Tracker in the zxtune; Bcuz the modland only has 12 .sng of Sam Coupe E-Tracker, & it won't worth it.

**Comment by ciros88** [comment]
im not sure what those .sng files are but evil.sng it is Sam Coupe E-Tracker and works in zxtune (thus in bzr2 if removing/renaming the webuade plugin dll #583 )

**Comment by rayman3003** [comment]
> im not sure what those .sng files are

Yeap. It could be possible that modland put them in a wrong folder.

I searched github, & found these 3 playback library for Sam Coupe.

If possible, please check them out; & if none of them could play these sng files, it's better close this issue.

https://github.com/Deltafire/SCPlayer

https://github.com/stefandrissen/e-tracker-player

https://github.com/bushy555/Sam_Coupe_1-bit_Music

**Comment by ciros88** [comment]
> thanks to [#91](https://github.com/aargirakis/BZRPlayer/issues/91) now EVIL.SNG works but is misdetected by webuade first (see [#583](https://github.com/aargirakis/BZRPlayer/issues/583))

with the upcoming matti's uade fork adoption this is not misdetected by uade anymore and can finally play (backed by zxtune)

**Comment by ciros88** [comment]
MUSIC1.SNG & MUSIC2.SNG are here: https://modland.com/pub/modules/Sam%20Coupe%20SNG/Frantisek%20Fuka

**Comment by rayman3003** [comment]
Some sources that might help for this playback issue:

https://github.com/mdsitton/SngFileFormat
https://github.com/joshrmcdaniel/sng-format-python
